### PR TITLE
chore: Adds a test_name param to acceptance test workflow

### PR DESCRIPTION
**Pull request by maastha**
## Description

This PR enhances the Acceptance test workflow by enabling users to specify and run a single test within a Test Group. 

This feature is particularly beneficial for executing long-running tests (such as cluster tests) during development, allowing developers to quickly test changes without waiting for the entire test suite/group to complete or tying up local machine resources.

Additionally, this helps in avoiding consuming more resources than strictly necessary for some feature/bug that we might be working on. 

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Further comments
Test run for below workflow: https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9567292550
Screenshot 2024-06-18 at 15 45 06

### CODE REVIEW

File: .github/workflows/acceptance-tests-runner.yml [hunk 1]
```diff
@@ -133,7 +137,7 @@ env:
   TF_LOG: ${{ vars.LOG_LEVEL }}
   ACCTEST_TIMEOUT: ${{ vars.ACCTEST_TIMEOUT }}
   # Only Migration tests are run when testing a specific previous provider version instead of the latest one
-  ACCTEST_REGEX_RUN: ${{ inputs.provider_version == '' && '^Test(Acc|Mig)' || '^TestMig' }}  
+  ACCTEST_REGEX_RUN: ${{ inputs.provider_version == '' && '^TestAccNetworkRSNetworkPeering_basicAzure' || '^TestMig' }}
```
Comment (oarbusi) [inline .github/workflows/acceptance-tests-runner.yml]: looks like this was a temporary change?

File: .github/workflows/acceptance-tests-runner.yml [hunk 2]
```diff
@@ -653,13 +664,9 @@ jobs:
           AZURE_VNET_NAME: ${{ secrets.azure_vnet_name }}
           AZURE_VNET_NAME_UPDATED: ${{ secrets.azure_vnet_name_updated }}
           MONGODB_ATLAS_LAST_VERSION: ${{ needs.get-provider-version.outputs.provider_version }}
+          ACCTEST_REGEX_RUN: ${{ inputs.test_name }}
           ACCTEST_PACKAGES: |
-            ./internal/service/networkcontainer
             ./internal/service/networkpeering
-            ./internal/service/privateendpointregionalmode
```
Comment (oarbusi) [inline .github/workflows/acceptance-tests-runner.yml]: I guess removing this was supposed to be temporary?

File: .github/workflows/acceptance-tests-runner.yml [hunk 3]
```diff
@@ -246,12 +250,7 @@ jobs:
             - 'internal/service/ldapconfiguration/*.go'
             - 'internal/service/ldapverify/*.go'  
           network:
-            - 'internal/service/networkcontainer/*.go'
```
Comment (oarbusi) [inline .github/workflows/acceptance-tests-runner.yml]: same with these removals

Comment (maastha) [reply .github/workflows/acceptance-tests-runner.yml]: aah sorry nice catch, these changes sneaked in from my previous branch, let me clean up
Thank you!!

**Review by oarbusi — APPROVED** [review]
LGTM once temporary changes are removed! Nice work

File: .github/workflows/acceptance-tests-runner.yml (hunk 1 above)
Comment (maastha) [reply .github/workflows/acceptance-tests-runner.yml]: fixed, resolving comment, feel free to reopen

File: .github/workflows/acceptance-tests-runner.yml (hunk 3 above)
Comment (maastha) [reply .github/workflows/acceptance-tests-runner.yml]: fixed, resolving comment, feel free to reopen

**Pull request merged**